Add touchscreen support

JoshHumpherys opened the pull request:
Adds touchscreen support for phones, tablets, touchscreen laptops, etc.

Note: I added babel-core to package.json. Also, tapping twice in chrome zooms (it thinks we're double tapping so we might want to disable that in the future).

SyntaxBlitz commented:
Hi!!

This PR is still on my radar. I was putting it off because I thought my dev environment was messed up -- mostly because I couldn't remember how the title got changed to "PLL trainer" and I assumed I accidentally crossed branches in version control -- but actually I think everything's fine.

I'm not sold on the navbar changes -- it looks a bit out of place with the gradient inverted at the top. I'm open to new ideas here, but it should flow better with the design.

I also notice that (on my mobile Chrome build) a single tap will change the alg twice, once immediately (from touchStart) and then another after 300ms (from mouseDown). I think the best option here is to use just mouseDown, and (if you don't like the 300ms delay) use the [fancy meta tags](https://www.telerik.com/blogs/what-exactly-is.....-the-300ms-click-delay) to remove the delay.

Please make these changes before we merge. Or don't. I don't blame you if you don't want to touch this repo again, a full year after your initial PR :). Feel free to close out the PR, in that case.

JoshHumpherys commented:
@SyntaxBlitz No worries on the late turn-around. I forked the repo about a year ago, and added touchscreen support just for me to use for the few weeks I was learning OLL. I only spent about an hour or so hacking up the original implementation, but I figured I might as well open a PR.

I fixed the navbar, it was hideous, I agree. Really not sure what led one-year-younger me to think that looked good :joy:. Still doesn't look great, so feel free to push back on it, but it's significantly improved from the reverse gradient :man_facepalming:.

And yeah, the 300ms delay is infuriating, but it appears to me to be slower than 300ms. I'm almost wondering if it's lagging on mobile for some reason. I'm not planning on investigating it further though.

SyntaxBlitz commented:
Good work -- this is definitely better! I still have a couple nitpicks:

- let's use `#7AB317` on the navbar background and a `rgba(0, 0, 0, 0.2)` or similar for the box-shadow
- looks like the navbar causes an issue when you scroll all the way to the bottom of the OLL selector page
- clicking on the "Press space or click here to generate an OLL case." text now double-fires the generator, so let's remove the event handler on that `<p>` tag.

I can also make these changes if you don't want to bother updating the PR. Let me know.
